digitalocean/reservedipv6/resource_reserved_ipv6_test.go

```diff
@@ -0,0 +1,179 @@
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      testAccCheckDigitalOceanReservedIPV6Destroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckDigitalOceanReservedIPV6Config_droplet(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDigitalOceanReservedIPV6Exists("digitalocean_reserved_ipv6.foobar", &reservedIPv6),
+					resource.TestCheckResourceAttr(
+						"digitalocean_reserved_ipv6.foobar", "region_slug", "nyc3"),
+				),
+			},
+			{
+				Config: testAccCheckDigitalOceanReservedIPv6Config_Reassign(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDigitalOceanReservedIPV6Exists("digitalocean_reserved_ipv6.foobar", &reservedIPv6),
+					resource.TestCheckResourceAttr(
+						"digitalocean_reserved_ipv6.foobar", "region_slug", "nyc3"),
+				),
+			},
+			{
+				Config: testAccCheckDigitalOceanReservedIPV6Config_Unassign(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDigitalOceanReservedIPV6Exists("digitalocean_reserved_ipv6.foobar", &reservedIPv6),
+					resource.TestCheckResourceAttr(
+						"digitalocean_reserved_ipv6.foobar", "region_slug", "nyc3"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckDigitalOceanReservedIPV6Destroy(s *terraform.State) error {
+	client := acceptance.TestAccProvider.Meta().(*config.CombinedConfig).GodoClient()
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "digitalocean_reserved_ipv6" {
+			continue
+		}
+
+		// Try to find the key
+		_, _, err := client.ReservedIPV6s.Get(context.Background(), rs.Primary.ID)
+
+		if err == nil {
+			return fmt.Errorf("Reserved IPv6 still exists")
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckDigitalOceanReservedIPV6Exists(n string, reservedIPv6 *godo.ReservedIPV6) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No Record ID is set")
+		}
+
+		client := acceptance.TestAccProvider.Meta().(*config.CombinedConfig).GodoClient()
+
+		// Try to find the ReservedIPv6
+		foundReservedIP, _, err := client.ReservedIPV6s.Get(context.Background(), rs.Primary.ID)
+
+		if err != nil {
+			return err
+		}
+
+		if foundReservedIP.IP != rs.Primary.ID {
+			return fmt.Errorf("Record not found")
+		}
+
+		*reservedIPv6 = *foundReservedIP
+
+		return nil
+	}
+}
+
+var testAccCheckDigitalOceanReservedIPV6Config_regionSlug = `
+resource "digitalocean_reserved_ipv6" "foobar" {
+  region_slug = "nyc3"
+}`
+
+func testAccCheckDigitalOceanReservedIPV6Config_droplet(name string) string {
+	return fmt.Sprintf(`
+resource "digitalocean_droplet" "foobar" {
+  name               = "%s"
+  size               = "s-1vcpu-1gb"
+  image              = "ubuntu-22-04-x64"
+  region             = "nyc3"
+  ipv6               = true
+  private_networking = true
```

Comment:
I'm seeing an error from terraform.  Let's remove `private_networking` from this example.
```console
│ Warning: Argument is deprecated
│ 
│   with digitalocean_droplet.foobar,
│   on main.tf line 35, in resource "digitalocean_droplet" "foobar":
│   35:   private_networking = true
│ 
│ This parameter has been deprecated. Use `vpc_uuid` instead to specify a VPC network for the Droplet. If no `vpc_uuid` is provided, the Droplet will be placed in your account's
│ default VPC for the region.
│ 
```